tst.py

```diff
@@ -1670,6 +1670,21 @@ class C:
         # Make sure we're still talking to our custom mapping.
         self.assertEqual(fields(C)['i'].metadata['xyzzy'], 'plugh')
 
+    def test_generic_dataclasses(self):
+        T = TypeVar('T')
+
+        @dataclass
+        class LabeledBox(Generic[T]):
+            content: T
+            label: str = '<unknown>'
+
+        box = LabeledBox(42)
+        self.assertEqual(box.content, 42)
+        self.assertEqual(box.label, '<unknown>')
+
+        # subscripting the resulting class should work, etc.
+        boxes: List[LabeledBox[int]] = []
```

Comment:
There is a caveat: variable annotations at function/method scope a never evaluated at runtime, so that this test will always pass (as well as ``x: 1/0 = 5`` at a function scope). Therefore this test should be like this:
```python
Alias = List[LabeledBox[int]]
```
```

Reply:
Of course. Thanks.